bin/katana/src/cli/init/mod.rs

```diff
@@ -0,0 +1,330 @@
+use std::fmt::Display;
+use std::fs;
+use std::path::PathBuf;
+use std::str::FromStr;
+use std::sync::Arc;
+
+use anyhow::{anyhow, Context, Result};
+use cainome::rs::abigen;
+use clap::Args;
+use dojo_utils::TransactionWaiter;
+use inquire::{Confirm, CustomType, Text};
+use katana_cairo::lang::starknet_classes::casm_contract_class::CasmContractClass;
+use katana_cairo::lang::starknet_classes::contract_class::ContractClass;
+use katana_primitives::{felt, ContractAddress, Felt};
+use serde::{Deserialize, Serialize};
+use starknet::accounts::{Account, ConnectedAccount, ExecutionEncoding, SingleOwnerAccount};
+use starknet::contract::ContractFactory;
+use starknet::core::types::contract::{CompiledClass, SierraClass};
+use starknet::core::types::{BlockId, BlockTag, FlattenedSierraClass, StarknetError};
+use starknet::core::utils::{cairo_short_string_to_felt, parse_cairo_short_string};
+use starknet::providers::jsonrpc::HttpTransport;
+use starknet::providers::{JsonRpcClient, Provider, ProviderError, Url};
+use starknet::signers::{LocalWallet, SigningKey};
+use tokio::runtime::Runtime;
+
+#[derive(Debug)]
+struct InitInput {
+    /// the account address that is used to send the transactions for contract
+    /// deployment/initialization.
+    account: ContractAddress,
+
+    // the id of the new chain to be initialized.
+    id: String,
+    // the chain id of the settlement layer.
+    l1_id: String,
```

Comment:
should we refer to this as `settlement_` rather than `l1_`? since starknet is a l2?